ee/kernel/src/sifcmd.c

```diff
@@ -155,19 +158,14 @@ int _SifCmdIntHandler(int channel)
 
     if (header->cid & SIF_CMD_ID_SYSTEM) {
         if ((u32)id < cmd_data->nr_sys_handlers)
-            cmd_handlers = cmd_data->sys_cmd_handlers;
-        else
-            goto out;
+            if (cmd_data->sys_cmd_handlers[id].handler)
```

Comment:
this function changed its logic, is that right?

Reply:
The function behavior is the same. However, changes were needed due to the fact that the structure types are now different (due to the added member)